osu.Game/Beatmaps/Timing/TimeSignature.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Beatmaps.Timing
+{
+    /// <summary>
+    /// Stores the time signature of a track.
+    /// For now, the lower numeral can only be 4; support for other denominators can be considered at a later date.
+    /// </summary>
+    public class TimeSignature : IEquatable<TimeSignature>
```

Comment:
I purposefully did not use readonly struct. You would be able to then do `new TimeSignature()` which would be zero initialised and therefore invalid. And you can neither disallow that or add default values to a readonly struct.

Reply:
The point about invalidly zero initialised and also not have default values sounds fair. You might as well make `SimpleTriple` and `SimpleQuadruple` statics initialised once rather than per access.

Reply:
change about initialisation applied in 4e5a1f2